docs/en/operations/settings/settings.md

```diff
@@ -404,6 +404,32 @@ Possible values:
 
 Default value: 0.
 
+## any_join_distinct_right_table_keys {#any_join_distinct_right_table_keys}
+
+Enables or disables specific ClickHouse server behavior in `ANY INNER|LEFT|RIGHT| JOIN` operations.
+
+When the behavior enabled:
+
+- Results of `t1 ANY LEFT JOIN t2` and `t2 ANY RIGHT JOIN t1` operations are not equal because ClickHouse uses the logic with many-to-one left-to-right table keys mapping.
+- Results of `ANY INNER JOIN` operations contain all rows from the left table like the `SEMI LEFT JOIN` operations do.
+
+When the behavior disabled:
```

Comment:
It's better do discribe it first cause this setting is backward compatibility. So it should be disabled in future versions and removed some day.